### PR TITLE
New Port Goblin Dungeoneer

### PR DESCRIPTION
**Pull request by Fraxinus88**
# New Port for Goblin Dungeoneer

## Game Information
- **Title**: Goblin Dungeoneer
- **URL**: https://crackedghostgames.itch.io/goblin-dungeoneer

### CFW Tests
[X] AmberELEC
[X] ArkOS
[X] JELOS/ROCKNIX
[X] MuOS

Architecture
[X] armHF
[X] Panfrost (Mainline)

Resolution Tests
Test all major resolutions:
[X] 480x320
[X] 640x480
[X] 720x720 (RGB30)
[X] Higher resolutions (e.g., 1280x720

## File Structure
- Your port should have the following structure:
  - portname/
    - port.json
    - README.md
    - screenshot.jpg
    - cover.jpg
    - gameinfo.xml
    - Port Name.sh
    - portname/
      - <portfiles here>

## Additional Resources
For an in-depth guide on creating a pull request, refer to: [PortMaster Game Packaging Guide](https://portmaster.games/packaging.html#creating-a-pull-request)

### CODE REVIEW

**Review by Slayer366 — DISMISSED** [review]

File: ports/goblindungeoneer/Goblin Dungeoneer.sh [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+elif [ -d "/opt/tools/PortMaster/" ]; then
+  controlfolder="/opt/tools/PortMaster"
+elif [ -d "$XDG_DATA_HOME/PortMaster/" ]; then
+  controlfolder="$XDG_DATA_HOME/PortMaster"
+else
+  controlfolder="/roms/ports/PortMaster"
+fi
+
+source $controlfolder/control.txt
+source $controlfolder/device_info.txt
```
Comment (Cebion) [inline ports/goblindungeoneer/Goblin Dungeoneer.sh:16]:
```suggestion
```

File: ports/goblindungeoneer/Goblin Dungeoneer.sh [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+export PORT_32BIT="Y"
+
+get_controls
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
+
+$ESUDO chmod 666 /dev/tty0
```
Comment (Cebion) [inline ports/goblindungeoneer/Goblin Dungeoneer.sh:22]:
```suggestion
```

File: ports/goblindungeoneer/Goblin Dungeoneer.sh [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        exit 1
+    fi
+else    
+    echo "Error: Missing files in gamedata folder or game has been patched."
+fi
+
+$GPTOKEYB "gmloader" &
+
+$ESUDO chmod +x "$GAMEDIR/gmloader"
+
```
Comment (Cebion) [inline ports/goblindungeoneer/Goblin Dungeoneer.sh:72]:
```suggestion
pm_platform_helper "$GAMEDIR/gmloader"
```

File: ports/goblindungeoneer/Goblin Dungeoneer.sh [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+./gmloader goblindungeoneer.apk
+
+$ESUDO kill -9 $(pidof gptokeyb)
```
Comment (Cebion) [inline ports/goblindungeoneer/Goblin Dungeoneer.sh:75]:
```suggestion
```

File: ports/goblindungeoneer/Goblin Dungeoneer.sh [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+$ESUDO systemctl restart oga_events &
```
Comment (Cebion) [inline ports/goblindungeoneer/Goblin Dungeoneer.sh:76]:
```suggestion
pm_finish
```

**Review by Cebion — CHANGES_REQUESTED** [review]

**Comment by Fraxinus88** [comment]
Changes committed :)

File: ports/goblindungeoneer/goblindungeoneer/patch/Patch Notes.txt
Comment (JanTrueno) [inline ports/goblindungeoneer/goblindungeoneer/patch/Patch Notes.txt:1]: Gotta be somewhat careful with adding original code in patchnotes, but this is simple enough thats its probably fine.

**Review by JanTrueno — APPROVED** [review]

**Review by Cebion — APPROVED** [review]

**Pull request merged**